source/common/filesystem/posix/filesystem_impl.cc

```diff
@@ -28,17 +28,38 @@ FileImplPosix::~FileImplPosix() {
   }
 }
 
-void FileImplPosix::openFile() {
-  const int flags = O_RDWR | O_APPEND | O_CREAT;
-  const int mode = S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH;
-
-  fd_ = ::open(path_.c_str(), flags, mode);
+void FileImplPosix::openFile(FlagSet in) {
+  const auto flags_and_perm = translateFlag(in);
```

Comment:
You reference 'perm' in this PR a few times; can we spell that out? I think it might mean "permanent" but I'm not sure.

Reply:
You mean move this into shared interface .  It seems contradictory to previous changes, see 
 https://github.com/envoyproxy/envoy/pull/7445#discussion_r302174360

Reply:
Oh it's hard to see exactly what Matt had an issue with because you squashed commits. In general please try to avoid that because we lose the comment history.

I still don't understand the meaning of 'perm'. But looking at how it's used here I retract my "make part of the interface" suggestion, and instead suggest using the correct type and name in posix for the 3rd arg to open: `mode_t mode`